### PR TITLE
add yamllint

### PR DESCRIPTION
**Pull request by irisdingbj**
add yamllint check for prow only
fix encoding issues for some yaml files

### CODE REVIEW

File: bin/yamllintrules [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  /tests/e2e/tests/pilot/testdata/networking/v1alpha3/rule-503test-destinationrule-c.yaml
+  /tests/e2e/tests/pilot/testdata/networking/v1alpha3/destination-rule-c.yaml
+  /tests/e2e/tests/pilot/testdata/networking/v1alpha3/rule-503test-destinationrule-c-add-subset.yaml
+  /tests/e2e/tests/pilot/testdata/networking/v1alpha3/rule-503test-destinationrule-c-del-subset.yaml
+  /tests/e2e/tests/pilot/testdata/networking/v1alpha3/destination-rule-c-private.yaml
+  /galley/tools/gen-meta/metadata.yaml
+  /galley/metadata.yaml
+
+rules:
+  braces: disable
```
Comment (howardjohn) [inline bin/yamllintrules]: why is pretty much every rule disabled?

**Comment by howardjohn** [comment]
We should probably add this to prow as we are trying to turn down our circleci usage. See https://github.com/istio/istio/commit/3dcda1e9e222681059af691a0f1d081eca0c81b6

To do this youll need to add a script in `prow/...` and add it to https://github.com/istio/test-infra/pull/1316

I think it would be better if you just added this to the bin/linters script though, we already have a series of linters that run in that anyways. Then you just need to run `make lint` and you know everything is linted

File: .circleci/config.yml [hunk 2]
```diff
@@ -381,6 +381,7 @@ jobs:
       - checkout
       - run: make sync
       - run: make lint
+      - run: make yamllint
```
Comment (howardjohn) [inline .circleci/config.yml]: oh actually didn't notice this was just added to the lint job not a new job, so nevermind.

It still could just be in `make lint` though :slightly_smiling_face: 

Looks like the image doesn't have yamllint though: `bin/yaml_lint.sh: line 33: yamllint: command not found`

**Comment by linsun** [comment]
Agreed with @howardjohn we should move this to prow as well.

**Comment by howardjohn** [comment]
we can add this to lint target and then move the whole linter to prow
ideally

**Comment by howardjohn** [comment]
https://github.com/istio/istio/pull/14934 adds lint to prow

File: bin/yamllintrules (hunk 1 above)
Comment (irisdingbj) [reply bin/yamllintrules]: Will change these files under related rules.

Comment (irisdingbj) [reply bin/yamllintrules]: > Will change these files under related rules.

The above files are actually template files so they all will fail the syntax check, so we need to ignore them all. for the metadata.yaml I have put it into the related rules.

**Comment by irisdingbj** [comment]
@howardjohn @sdake Can you review the new change here? Thanks

File: bin/yaml_lint.sh [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+set -e
+
+SCRIPTPATH=$( cd "$(dirname "$0")" ; pwd -P )
+
+ROOTDIR=$SCRIPTPATH/..
+cd "$ROOTDIR"
+
+PKGS=${PKGS:-"."}
+if [[ -z ${YAML_FILES} ]];then
+    YAML_FILES=$(find "${PKGS}" -type f -name '*.yaml' | grep -v ./vendor | grep -v -E "\/templates\/" | grep -v -E "template")
```
Comment (howardjohn) [inline bin/yaml_lint.sh]: interesting, I was wondering how you do the templated ones. We have `helm lint` which will at least catch some issues on those though so not a big concern

File: install/kubernetes/helm/istio/charts/grafana/values.yaml [hunk 4]
```diff
@@ -64,24 +64,24 @@ datasources:
   datasources.yaml:
     apiVersion: 1
     datasources:
-    - name: Prometheus
```
Comment (howardjohn) [inline install/kubernetes/helm/istio/charts/grafana/values.yaml]: was this failing the linter? I think we use this indentation style pretty often

Comment (irisdingbj) [reply install/kubernetes/helm/istio/charts/grafana/values.yaml]: Yes, it faild under current rules.

File: bin/yaml_lint.sh (hunk 3 above)
Comment (irisdingbj) [reply bin/yaml_lint.sh]: Currently ignored most of the template files.

**Pull request merged**